PayItForward.Web/Controllers/HomeController.cs

```diff
@@ -3,31 +3,41 @@
     using System;
     using System.Collections.Generic;
     using System.Diagnostics;
+    using System.Linq;
+    using System.Security.Claims;
     using AutoMapper;
     using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
     using Microsoft.AspNetCore.Mvc;
+    using PayItForward.Common;
     using PayItForward.Services.Abstraction;
     using PayItForward.Web.Models;
     using PayItForward.Web.Models.CategoryViewModels;
     using PayItForward.Web.Models.HomeViewModels;
     using PayItForward.Web.Models.StoryViewModels;
 
-    [AllowAnonymous]
     public class HomeController : Controller
     {
         private const int ItemsPerPage = 3;
 
         private readonly IStoriesService storiesService;
         private readonly ICategoriesService categoriesService;
         private readonly IMapper mapper;
+        private readonly IHttpContextAccessor httpaccessor;
 
-        public HomeController(IStoriesService storiesService, ICategoriesService categoriesService, IMapper mapper)
+        public HomeController(
+            IStoriesService storiesService,
+            ICategoriesService categoriesService,
+            IMapper mapper,
+            IHttpContextAccessor httpaccessor)
         {
             this.storiesService = storiesService;
             this.categoriesService = categoriesService;
             this.mapper = mapper;
+            this.httpaccessor = httpaccessor;
         }
 
+        [AllowAnonymous]
```

Comment:
You dont need Allow anonymous on the action when the controller does not have authorize